scipy/optimize/_numdiff.py

```diff
@@ -344,34 +365,81 @@ def fun_wrapped(x):
     if np.any((x0 < lb) | (x0 > ub)):
         raise ValueError("`x0` violates bound constraints.")
 
-    h = _compute_absolute_step(rel_step, x0, method)
+    if as_linear_operator:
+        if rel_step is None:
+            rel_step = _compute_relative_step(method)
+
+        return _linear_operator_difference(fun_wrapped, x0,
```

Comment:
I made some tests and indeed it does make a lot of difference in the result.

I created an experiment ([source code](https://gist.github.com/antonior92/67257d3edf361c6592169e817acf2d9e)) where ``p`` could assume very large and very small values:
```
p = exp(u), for u = random.uniforme(lb=-15, ub=15)
```

When using ``rel_step`` the mean percentual errors were:
```
2-point error = 0.00242135098928
3-point error = 0.0719997740106
cs error = 2.30640531383e-07
```
using ``rel_step/norm(p)`` I got:
```
2-point error = 2.59029784232e-07
3-point error = 3.49325916396e-10
cs error = 1.07554062691e-16
```